scripts/data.py

```diff
@@ -0,0 +1,216 @@
+    """
+    resp = requests.post(f"{root_api_endpoint()}/workspaces/{workspace}")
+
+    if resp.ok:
+        return True
+
+    return False
+
+
+def create_graph(
+    workspace: str, graph_name: str, node_tables: List[str], edge_table: str
+) -> bool:
+    """Create a graph."""
+    resp = requests.post(
+        f"{root_api_endpoint()}/workspaces/{workspace}/graphs/{graph_name}",
+        params={"node_tables": node_tables, "edge_table": edge_table},
+    )
+
+    if resp.ok:
+        return True
+
+    return False
+
+
+def table_exists(workspace: str, table: str) -> bool:
+    """Check if the table exists."""
+    resp = requests.get(f"{root_api_endpoint()}/workspaces/{workspace}/tables/{table}")
+
+    if resp.status_code == 400:
+        return True
+    return False
+
+
+def create_table(workspace: str, table: str, data: str) -> bool:
+    """
+    Create table.
+
+    Returns True if successful, False otherwise
+    """
+    resp = requests.post(
+        f"{root_api_endpoint()}/csv/{workspace}/{table}", data=data.encode("utf-8")
+    )
+
+    if resp.ok:
+        return True
+    return False
+
+
+def log(text: str, indent: int):
+    """Log to console output."""
+    click.echo(click.wrap_text(text, initial_indent=(indent * " ")))
+
+
+@click.group()
+def cli():
+    """Script that helps with bootstrapping example data."""
+    pass
+
+
+@cli.command("populate")
+@click.argument("address", nargs=1, required=False)
+def populate(address: str):
+    """Populate arangodb with example data."""
+    global server_address
+    log_tabstop = 4
+    log_indent = 0
+
+    if address is not None:
+        address_parts = address.split(":")
+        address_parts = [x for x in address_parts if x]
+
+        if len(address_parts) != 2:
+            raise Exception("Invalid passed address.")
+
+        server_address = address
+
+    log(f"Populating data on {server_address}...", indent=log_indent)
+
+    for path in DATA_DIR.iterdir():
+        workspace = path.name
+
+        log(f'Processing dataset "{workspace}"', indent=log_indent)
+        log_indent += log_tabstop
+
+        files = tuple(path.glob("*.csv"))
+
+        try:
+            if check_workspace_exists(workspace):
+                log(
+                    f'Workspace "{workspace}" already exists, skipping...',
+                    indent=log_indent,
+                )
+                continue
+        except requests.exceptions.ConnectionError:
+            raise Exception(f"\tConnection could not be established at {address}")
+
+        if not create_workspace(workspace):
+            log(f"Error creating workspace {workspace}.", indent=log_indent)
+            continue
+
+        # Create Tables
+        for file in files:
+            table_name = file.stem
+
+            if table_exists(workspace, table_name):
+                log(
+                    f'Table "{table_name}" already exists, skipping...',
+                    indent=log_indent,
+                )
```

Comment:
The use case we're targeting is that this very script created the workspace, which means it should be empty. If this error condition occurs, something has gone very wrong, and we should abort with a fatal error message, rather than simply skip.

Reply:
Good point. I'll change that.